CIP-0118/README-nested.md

```diff
@@ -0,0 +1,538 @@
+unbalanced transactions are required. Such a service can be run by anyone, probably 
+alongside a full node. It would have to have a filter set for the kinds of transactions 
+it will engage with. There is no guarantee that every unbalanced transaction 
+will eventually be balanced by a subsequent incoming one, so each service will have 
+to solve this problem for themselves, probably using an appropriate filter.
+
+#### Batch Observers
+
+A service processing underspecified transactions (such as the Babel service) may accept incoming transactions 
+that contain non-empty `requiredObservers`. This means this service must construct batches containing these 
+transactions and also satisfy the batch-level constraints imposed by the scripts specified in their `requiredObservers` 
+in order to create valid batches.
+
+An additional high-level language may be beneficial to specify what the batch observer scripts 
+actually require of the batch, as Plutus script constraints may be difficult to work with directly.
+
+#### Light Client Service
+
+The design outlined here can facilitate a trustless LC service.
+Light clients may use the `corInputs` and `spendOuts` features to execute the following protocol.
+LC is the light client, and SP is the service provider. We assume there exists a 
+high-level language in which a light client can specify what they want a 
+transaction constructed for them by the SP to do, e.g. pay `x` amount from my 
+wallet to some other wallet, or mint some NFT. 
+
+1. LC -> SP : a high-level specification `s` 
+2. SP : construct a `tx` that 
+- meets `s`, 
+- includes one input `i` from SP's wallet
+- includes a payment to SP for light-client services
+3. SP : replace all inputs except `i` in `tx` with corresponding `spendOuts` from the UTxO set to make `tx'`
+4. SP -> LC : `tx'`
+5. LC : checks `tx'` meets `s`, signs it
+6. LC -> SP : signed `tx'`
+7. SP : constructs top-level `tx''` that provides the `corInputs` for `tx'`
+8. SP : submits to Cardano the top-level `tx''` with sub-tx `tx'`
+
+Note that `tx'` may also be unbalanced, simultaneously allowing the LC 
+to make a payment in a non-Ada token for the service. This design ensures that 
+an LC cannot construct a fully valid transaction from `tx'`, since they do not have access to 
+the current UTxO (presumably, if they did, they would construct their own valid transaction).
+It also ensures that the SP is paid (in step 2) for their services. 
+
+## Rationale: how does this CIP achieve its goals?
+
+Same as the related CIPs/ motivation. More info here later. 
+
+## Use cases
+
+### Open (atomic) swaps
+
+A user wants to swap 10 Ada for 5 tokens `myT`. He creates an unbalanced transaction `tx` that 
+has extra 10 Ada, but is short 5 `myT`. 
+Any counterparty that sees this transaction can create a top-level 
+transaction `tx'` that includes `tx` as a sub-transaction. The transaction 
+`tx'` would have extra 5 `myT`, and be short 10 Ada. 
```

Comment:
Can you possibly be a bit more explicit with this example? This CIP talks about underspecified transactions, but doesn't really give a user-friendly example of one. I believe this new version has adopted the approach from my [Transaction Pieces](https://github.com/cardano-foundation/CIPs/pull/873), right? So it would be accurate to say:
- `tx` has inputs with 10 Ada and an output with 5 `myT`, and is left unbalanced like this.
- `tx'` has inputs with 5 `myT` and an output with 10 Ada.
- They are submitted together as a balanced batch with `tx'` as the top-level transaction.

I think I am biased by the original Validation Zones CIP that had unresolved holes, so I would personally find this example helpful to show there are no unresolved holes in this version.

Reply:
yes, this is exactly right.